nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/permissions/test/TestCommentManager.java

```diff
@@ -0,0 +1,108 @@
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.platform.comment.api.CommentManager;
+import org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CommentFeature.class })
+public class TestCommentManager {
+
+    @Inject
+    protected CoreSession systemSession;
+
+    @Inject
+    protected CommentManager commentManager;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void testCreateReadDelete() throws Exception {
+        // Create a user
+        DocumentModel userModel = userManager.getBareUserModel();
+        String schemaName = userManager.getUserSchemaName();
+        userModel.setProperty(schemaName, "username", "Foo");
+        userModel = userManager.createUser(userModel);
+
+        // Create a folder
+        DocumentModel folder = systemSession.createDocumentModel("/", "folder", "Folder");
+        folder = systemSession.createDocument(folder);
+
+        // Create a document
+        DocumentModel doc = systemSession.createDocumentModel("/folder/", "TestFile", "File");
+        doc = systemSession.createDocument(doc);
+
+        // Set The right ACE so Foo can read the document
+        ACP acp = doc.getACP();
+        ACE grantRead = new ACE("Foo", SecurityConstants.READ, true);
+        acp.addACE(ACL.LOCAL_ACL, grantRead);
+        doc.setACP(acp, true);
+
+        try (CoreSession userSession = coreFeature.openCoreSession("Foo")) {
+            // Get the document as the Foo user
+            DocumentModel userDoc = userSession.getDocument(doc.getRef());
+
+            // Comment the document as Foo
+            DocumentModel comment = userSession.createDocumentModel(CommentsConstants.COMMENT_DOC_TYPE);
+            comment.setProperty("comment", "text", "This is my comment");
+            commentManager.createComment(userDoc, comment);
+
+            // Check the comment document can be retrieved by a system session query
+            List<DocumentModel> dml = systemSession.query("SELECT * FROM " + CommentsConstants.COMMENT_DOC_TYPE);
+            assertEquals(1, dml.size());
+
+            // Check the comment document cannot be retrieved by a user session query
+            dml = userSession.query("SELECT * FROM " + CommentsConstants.COMMENT_DOC_TYPE);
+            assertEquals(0, dml.size());
+
+            // Check the comment can be retrieved by Foo via the comment service
+            List<DocumentModel> comments = commentManager.getComments(userDoc);
+            assertEquals(1, comments.size());
```

Comment:
Worth checking you actually get the comment you created before, by at least checking the comment text maybe?

Reply:
I added a check by retrieving the property that was set.
I didn't find an xpath builder to slug an array of words with ":" so I built the xpath by hand